### DISCUSSION TITLE
Coding Standard

### DISCUSSION THREAD

**Discussion by SkyeHoefling** [dnnsoftware/Dnn.Platform#3945] [General] [answered]
I am working on a pretty large change for Dependency Injection and have noticed inconsistencies across the `DotNetNuke.Library` in code formatting. There are also countless errors from stylecop to code that doesn't meet the stylecop defined guide. Do we have an adopted coding standard such as the [.NET Runtime Coding Standard](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md)? Is there a preference on using statements being inside the namespace or above the namespace declaration?

**Comment by valadas** [comment]
Yes, there are editor config files ans stylecop is setup, but since that is recent, we do have a lot of manual fixes to do. @bdukes did a lot of automated fixes but we now have more human ones to do.

**Comment by bdukes** [comment]
To your specific question, `using` statements should be inside the `namespace` declaration.

**Comment by bdukes** [comment, marked as answer]
#3810 has some more details on where the configuration comes from and a couple of notable adjustments from those baselines.

**Comment by SkyeHoefling** [comment]
Thanks for the links and answers 👍
